src/main/java/io/cryostat/recordings/ArchiveRequestGenerator.java

```diff
@@ -0,0 +1,277 @@
+                    }
+                });
+    }
+
+    @ConsumeEvent(value = ARCHIVE_ADDRESS)
+    public void onMessage(ArchiveRequest request) {
+        try {
+            performArchive(request);
+        } catch (Exception e) {
+            logger.warn("Exception thrown while servicing request: ", e);
+        }
+    }
+
+    @ConsumeEvent(value = GRAFANA_ARCHIVE_ADDRESS, blocking = true)
+    public void onMessage(GrafanaArchiveUploadRequest request) {
+        try {
+            logger.info("Job ID: " + request.getId() + " submitted.");
+            String result =
+                    recordingHelper
+                            .uploadToJFRDatasource(request.getPair())
+                            .await()
+                            .atMost(timeout);
+            logger.info("Grafana upload complete, firing notification");
+            bus.publish(
+                    MessagingServer.class.getName(),
+                    new Notification(GRAFANA_UPLOAD_SUCCESS, Map.of("jobId", request.getId())));
+        } catch (Exception e) {
+            logger.warn("Exception thrown while servicing request: ", e);
+            bus.publish(
+                    MessagingServer.class.getName(),
+                    new Notification(GRAFANA_UPLOAD_FAIL, Map.of("jobId", request.getId())));
+        }
+    }
+
+    @ConsumeEvent(value = GRAFANA_ACTIVE_ADDRESS, blocking = true)
+    public void onMessage(GrafanaActiveUploadRequest request) {
+        try {
+            logger.info("Job ID: " + request.getId() + " submitted.");
+            String result =
+                    recordingHelper
+                            .uploadToJFRDatasource(request.getTargetId(), request.getRemoteId())
+                            .await()
+                            .atMost(timeout);
+            logger.info("Grafana upload complete, firing notification");
+            bus.publish(
+                    MessagingServer.class.getName(),
+                    new Notification(GRAFANA_UPLOAD_SUCCESS, Map.of("jobId", request.getId())));
+        } catch (Exception e) {
+            logger.warn("Exception thrown while servicing request: ", e);
+            bus.publish(
+                    MessagingServer.class.getName(),
+                    new Notification(GRAFANA_UPLOAD_FAIL, Map.of("jobId", request.getId())));
+        }
+    }
+
+    @ConsumeEvent(value = ACTIVE_REPORT_ADDRESS, blocking = true)
+    public void onMessage(ActiveReportRequest request) {
+        try {
+            logger.info("Job ID: " + request.getId() + " submitted.");
+            Map<String, AnalysisResult> result =
+                    reportsService.reportFor(request.recording).await().atMost(timeout);
+            logger.info("Report generation complete, firing notification");
+            bus.publish(
+                    MessagingServer.class.getName(),
+                    new Notification(
+                            REPORT_SUCCESS, Map.of("result", result, "jobId", request.getId())));
```

Comment:
I *think* we can remove the `"result", result` from these notification Map payloads now. The `reportsService.reportFor()` call should be enough to seed the content into the caches, so that when the client sees the 202 response, tracks the ID, then sees this notification and follows up with the second request, the server can send the actual content again with a 200 response. So the websocket payload doesn't need to contain all this data, since the client isn't actually going to try to parse the analysis result out of the notification.

In theory the client *could* just get the response out of this notification and we could skip the "second request" part of the API flow, but since these websocket notifications get written out to every single connected client, I don't want them to contain unnecessary data. And it only happens to kind of make sense in this case because the response data is reasonable to serialize into JSON - if the response payload was some other content type (say it was a `.jfr` binary file) then the "second request" flow would still be required, and so I think that's the more generic solution.